### PR TITLE
add `exports` field and expose `/CustomAttributeRegistry` entrypoint

### PR DESCRIPTION
**Pull request by mayank99**
This adds an [`"exports"` field](https://nodejs.org/api/packages.html#package-entry-points) to the `package.json`, which is necessary for this package to be properly usable in an ESM environment.

Exposes two entrypoints, which can be used as follows:

```js
// all of these will execute side-effects from index.js
import "@lume/custom-attributes";
import { customAttributes } from "@lume/custom-attributes";
import { CustomAttributeRegistry } from "@lume/custom-attributes";
```

```js
// imports directly from dist/CustomAttributeRegistry.js
import { CustomAttributeRegistry } from "@lume/custom-attributes/CustomAttributeRegistry";
```

### CODE REVIEW

**Comment by trusktr** [comment]
Hello! Were you unable to `import { CustomAttributeRegistry } from "@lume/custom-attributes/dist/CustomAttributeRegistry.js"`? 

I'd like to avoid the Node package.json `exports` field because it is Node-specific, not something that browsers understand, and for that reason I would prefer that imports are the as similar as possible across envs (Node, Deno, browser with no build, build tools, or no build tools.

By adding the exports field, any other files automatically get blocked in Node.js, which means we would have to keep updating  the `exports` field if we add or modify the file structure. The simplest approach is to simply import the actual file paths.

Basically the goal is to keep imports as close to vanilla ESM as possible, avoiding any non-standard configs (Node exports is not part of ES spec), thus minimizing the amount of configuration needed for all ESM projects (browser or not).

I think you should be able to `import { CustomAttributeRegistry } from "@lume/custom-attributes/dist/CustomAttributeRegistry.js"` unless I missed something, but if not let me know. If so, then we can close this.

**Comment by trusktr** [comment]
In practice what the current approach means is that we can add the package to our vanilla ESM importmaps in this simple way:

```html
<script type="importmap">
{
  "imports": {
    "@lume/custom-attributes": "/node_modules/@lume/custom-attributes/dist/index.js",
    "@lume/custom-attributes/": "/node_modules/@lume/custom-attributes/"
  }
}
</script>
```

which allows

```js
// importing from the package name
import {...} from "@lume/custom-attributes"

// or importing from any subpath:
import {...} from "@lume/custom-attributes/dist/CustomAttributeRegistry.js"
```

Almost all of the Lume packages follow the same convention (a couple may be forked and not managed with `lume` CLI in which case they may differ), so if we were to make a change in the convention it would also have to be applied most of the packages here:

https://github.com/lume/lume/tree/develop/packages

**Comment by trusktr** [comment]
I'll go ahead and close this but if there's an issue I missed let me know! Thanks for the other two tweaks!
